### PR TITLE
feat: add sign in authentication 

### LINKED ISSUE #16: [FEATURE] <adding sign in route and generating jwt token>
### ⭐ Suggestion

Adding sign in page route and verifying the credentials of user with the data available on azure database. So, that the user is being allowed to sign in once the credentials is verified and also featuring auto generating tokens using jwt token, so we can verify the token with the user token and allowing to display personal details of the user once the token is verified.

### 💻 Use Cases

_No response_

### ❌ Related Problems

_No response_

### PR DESCRIPTION
**Pull request by chirag0002**
Implemented sign in authentication via verifying user credentials such as
- Username / E-mail
- Password
After sign in a unique token generated for after verification to be verified for displaying user personal information for which user has to be signed in (or we can say verified).

closes #16

### CODE REVIEW

File: src/News/News.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ axios.get("https://newsdata.io/api/1/news?apikey=pub_65940dd47f8c725cf5ef39fa683
 
   return (
     <div>
+    Newsssssssss
     {news.map((val) => (
```
Comment (SuperAayush) [inline src/News/News.js:17]: Remove this line of code (line :16)

File: src/middlewares/Checking.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    callCheckingPage();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div>checking</div>
+  )
+}
+
+export default Checking
```
Comment (SuperAayush) [inline src/middlewares/Checking.js:41]: Better to add comments to let users have an idea of what this file is about.

**Review by SuperAayush — CHANGES_REQUESTED** [review]

File: server/Routes/Routes.js [hunk 3]
```diff
@@ -51,6 +54,63 @@ Router.post("/register", async (req, res) => {
   }
 });
 
+
+// route for login using username or email
+
+Router.post('/signin', async (req,res) => {
```
Comment (chandankumar1307) [inline server/Routes/Routes.js:60]: Add a controller file and implement the router logic there. Keep Router.js clean with just the component

**Review by chandankumar1307 — CHANGES_REQUESTED** [review]
Restructure the code.

File: src/middlewares/Checking.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React, {useEffect} from 'react'
+import {useHistory} from 'react-router-dom'
+
+const Checking = () => {
+
+  const history = useHistory();
+
```
Comment (AayushShuk1a) [inline src/middlewares/Checking.js:7]: useHistory is deprecated please change it to useNavigate.

File: src/News/News.js (hunk 1 above)
Comment (chirag0002) [reply src/News/News.js:17]: Removed all unnecessary lines from the code.

File: src/middlewares/Checking.js (hunk 2 above)
Comment (chirag0002) [reply src/middlewares/Checking.js:41]: Comments are added to the files where needed.

File: server/Routes/Routes.js (hunk 3 above)
Comment (chirag0002) [reply server/Routes/Routes.js:60]: Implemented controllers to the routes.

File: src/middlewares/Checking.js (hunk 4 above)
Comment (chirag0002) [reply src/middlewares/Checking.js:7]: I think there is might updates in react-router-dom. useHistory is acclaimed and useNaviagte is deprecated. Please do notify me if there is something i missed.

Comment (AayushShuk1a) [reply src/middlewares/Checking.js:9]: useHistory was deprecated in 2019.In the latest version of react router dom useNavigate is better option to use.

**Review by chandankumar1307 — CHANGES_REQUESTED** [review]
Remove useHistory

File: src/middlewares/Checking.js (hunk 4 above)
Comment (chirag0002) [reply src/middlewares/Checking.js:9]: Yes, I have used useNavigate.

**Review by SuperAayush — APPROVED** [review]
LGTM

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good

**Review by bhavyastar — COMMENTED** [review]
LGTM

**Review by bhavyastar — APPROVED** [review]

**Review by chandankumar1307 — APPROVED** [review]
LGTM

**Pull request merged**